[ENG-382] Refactor python wrappers for coming enterprise features

arthurp opened the pull request:
**Breaking change**: All python programs now need to call `katana.local.initialize()` to setup the Katana runtime.

I chose `katana.local` a the package for the "shared memory" runtime. I'm open to other names, but I couldn't think of anything better and it aligns well with `katana.distributed`.

This PR does not include testing or error checking for what happens if a user tries to use katana without calling `initialize`. That will be needed. If waiting is acceptable I will create a tracking github issue. There is no clear place to put a TODO in the code since the issue affects every entry point to the katana Python API.

ddn0 commented on libtsuba/include/tsuba/Errors.h:
```diff
@@ -50,6 +51,8 @@ class ErrorCodeCategory : public std::error_category {
     case ErrorCode::AWSWrongRegion:
       return "AWS op may succeed in other region";
+    case ErrorCode::PropertyNotFound:
+      return "no such property";
```
Should we update the lines below to have the right error_condition for the error code too?

https://github.com/KatanaGraph/katana/pull/235/files#diff-3913e2030df8a04416b478e914e08a8cR63

tylershunt replied:
Ah yes. We totally should. and we will.

arthurp committed:
python: Refactor PropertyGraph.

Notable changes:

* PropertyGraph is no longer directly defined in property_graph
* The definition of PropertyGraph is not in a jinja file
* A separate jinja file is used for the numba wrappers
* References to PropertyGraph from Cython needed to be updated.
  References from Python do not need to change.
* Support for numba wrappers outside of the original definition
  module of the class was added.

arthurp committed:
python: Remove Cython implementation of BFS.

We don't support Cython user programming.

arthurp committed:
python: Refactor runtime "Sys" construction.

To initialize the shared memory runtime, the user must now call:

katana.local.initialize()

ddn0 commented on python/katana/_property_graph.pyx:
```diff
@@ -336,32 +274,83 @@ cdef class PropertyGraph:
+    @staticmethod
+    def from_graphml(path, uint64_t chunk_size=25000):
+        """
+        Load a GraphML file into katana form
```
qq: we still support GraphML?

arthurp replied:
I have no idea. I was asked to add this by @bowu and there was code implementing it in the graph-convert tools.

ddn0 reviewed (approved):
seems reasonable to me
